### PR TITLE
Added test for sandbox image

### PR DESCRIPTION
**Pull request by evalsocket**
# TL;DR
- Added github workflow for testing sandbox local image

 ## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue

fix: flyteorg/flyte#1474 & flyteorg/flyte#1108 & flyteorg/flyte#946

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

File: .github/workflows/functional-test.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+name: Functional test
+
+on:
+  pull_request:
+    paths:
+      - docker/**
+      - charts/**
```
Comment (kumare3) [inline .github/workflows/functional-test.yaml:7]: Should we add kustomize also?

Comment (evalsocket) [reply .github/workflows/functional-test.yaml:7]: kustomize is deprecated. Right ? It only validate sandbox end 2 end. In sandbox we don't use kustomize now

**Comment by evalsocket** [comment]
@eapolinario Do you have any suggestion for test script so that we can reuse it for both flyteorg and union

**Comment by evalsocket** [comment]
@eapolinario for now i removed your script and used currently available end 2 end test for testing our sandbox image. We will create separate PR

cc: @EngHabu

File: .github/workflows/functional-test.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+      - uses: whoan/docker-build-with-cache-action@v5.11.0
+        with:
+          image_name: flyteorg/flyte-sandbox
+          image_tag: dind
```
Comment (EngHabu) [inline .github/workflows/functional-test.yaml:27]: just to make it obvious, can you rename this to:
```suggestion
          image_tag: local
```

File: .github/workflows/functional-test.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+          registry: ghcr.io
+          build_extra_args: --target=dind --compress=true --build-arg=FLYTE_TEST=local
+          context: ./
+          dockerfile: docker/sandbox/Dockerfile
+          push_image_and_stages: false
+      - uses: unionai/flytectl-setup-action@v0.0.1
+      - name: Setup sandbox
+        run: |
+          flytectl sandbox start --image ghcr.io/flyteorg/flyte-sandbox:dind --imagePullPolicy Never
```
Comment (EngHabu) [inline .github/workflows/functional-test.yaml:36]:
```suggestion
          flytectl sandbox start --image ghcr.io/flyteorg/flyte-sandbox:local --imagePullPolicy Never
```

**Review by EngHabu — COMMENTED** [review]
A couple of nits

**Review by EngHabu — APPROVED** [review]

**Pull request merged**